src/main/java/org/dbos/apiary/xa/BitronixXAConnection.java

```diff
@@ -0,0 +1,138 @@
+package org.dbos.apiary.xa;
+
+import org.dbos.apiary.connection.ApiaryConnection;
+import org.dbos.apiary.function.FunctionOutput;
+import org.dbos.apiary.function.TransactionContext;
+import org.dbos.apiary.function.WorkerContext;
+import org.postgresql.util.PSQLException;
+import org.postgresql.util.PSQLState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.mysql.cj.jdbc.exceptions.MySQLTransactionRollbackException;
+
+import bitronix.tm.BitronixTransactionManager;
+import bitronix.tm.TransactionManagerServices;
+
+import java.lang.reflect.InvocationTargetException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class BitronixXAConnection extends XAConnection {
+
+    private static final Logger logger = LoggerFactory.getLogger(XAConnection.class);
+    public static String PostgresDBType = "Postgres";
+    public static String MySQLDBType = "MySQL";
+    BitronixXADBConnection pgConnection;
+    BitronixXADBConnection mqConnection;
+    BitronixTransactionManager btm = null;
+    @Override
+    public XADBConnection getXAConnection(String DBType) {
+        if (DBType.equals(PostgresDBType)) {
+            return this.pgConnection;
+        } else if (DBType.equals(MySQLDBType)) {
+            return this.mqConnection;
+        } else {
+            return null;
+        }
+    }
+    
+    public BitronixXAConnection(BitronixXADBConnection postgresConnection, BitronixXADBConnection mysqlConnection) throws SQLException {
```

Comment:
Since you can support two MySQL instances, the naming here should probably be just DB1 and DB2.